### PR TITLE
add baseUrl parameter to support slack-compatible integrations

### PR DESCRIPTION
**Pull request by andrzejwp**
Solution for #174 - additional configuration parameter to specify the full URL (before token) to support slack clones with compatible API, e.g. rocket.chat.

### CODE REVIEW

**Comment by chkal** [comment]
I'm not very familiar with Jenkins plugins, but as I'm interested in the feature, I just reviewed the PR. Looks very good to me. :+1:

File: src/main/java/jenkins/plugins/slack/SlackNotifier.java [hunk 1]
```diff
@@ -129,12 +134,13 @@ public String getCustomMessage() {
...
 
     @DataBoundConstructor
-    public SlackNotifier(final String teamDomain, final String authToken, final String room, final String buildServerUrl,
+    public SlackNotifier(final String baseUrl, final String teamDomain, final String authToken, final String room, final String buildServerUrl,
                          final String sendAs, final boolean startNotification, final boolean notifyAborted, final boolean notifyFailure,
                          final boolean notifyNotBuilt, final boolean notifySuccess, final boolean notifyUnstable, final boolean notifyBackToNormal,
                          final boolean notifyRepeatedFailure, final boolean includeTestSummary, CommitInfoChoice commitInfoChoice,
                          boolean includeCustomMessage, String customMessage) {
         super();
+        this.baseUrl = baseUrl;
```
Comment (samrocketman) [inline src/main/java/jenkins/plugins/slack/SlackNotifier.java]: Perhaps trim off potential spaces.

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java [hunk 2]
```diff
@@ -20,12 +22,14 @@
...
     private String host = "slack.com";
+    private String baseUrl;
     private String teamDomain;
     private String token;
     private String[] roomIds;
 
-    public StandardSlackService(String teamDomain, String token, String roomId) {
+    public StandardSlackService(String baseUrl, String teamDomain, String token, String roomId) {
         super();
+        this.baseUrl = baseUrl;
```
Comment (samrocketman) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java]: Trim off any potential spaces.  This should cover any case where `﻿StandardSlackService` is used.

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java [hunk 3]
```diff
@@ -38,7 +42,12 @@ public boolean publish(String message) {
     public boolean publish(String message, String color) {
         boolean result = true;
         for (String roomId : roomIds) {
+
             String url = "https://" + teamDomain + "." + host + "/services/hooks/jenkins-ci?token=" + token;
+            if (StringUtils.isEmpty(baseUrl)) {
```
Comment (samrocketman) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java]: What's the point of testing `StringUtils.isEmpty(baseUrl)` if it evaluates to true then use `baseUrl`?

**Comment by samrocketman** [comment]
:-1: for the issues I pointed out.  Other than that, looks good.

**Comment by andrzejwp** [comment]
Hi, I followed exactly how other parameters are handled. There is no trimming there, so I can either leave it without trimming, or we should trim all of them?
Thanks for the catch with the incorrect if() statement, I missed that.

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java (hunk 3 above)
Comment (andrzejwp) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java]: fixed, thanks

**Comment by samrocketman** [comment]
I would say correct the trimming associated with this change and then a second pull request can be created to address trimming all of the other values.  In general, I prefer to trim and sanitize human input as much as possible.

**Comment by Reamer** [comment]
:+1:  works. Thanks

**Comment by kmadel** [comment]
Needs help text. Also, and example in the README would be nice.

**Comment by MitchTalmadge** [comment]
👍 Any news on this?

**Comment by samrocketman** [comment]
I plan on performing a release this weekend. If you would like this PR to be in it then please rebase/resolve.  Also, please add help text for the option so users know what it is for.

**Comment by samrocketman** [comment]
This PR is superseded by #293.

**Pull request closed without merging**